src/core/meshcop/leader.hpp

```diff
@@ -26,11 +26,131 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if OPENTHREAD_MTD
-#include "leader_mtd.hpp"
-#elif OPENTHREAD_FTD
-#include "leader_ftd.hpp"
-#else
-#error "Please define OPENTHREAD_MTD=1 or OPENTHREAD_FTD=1"
-#endif
+/**
+ * @file
+ *   This file includes definitions for a MeshCoP Leader.
+ */
+
+#ifndef MESHCOP_LEADER_HPP_
+#define MESHCOP_LEADER_HPP_
+
+#include <coap/coap_client.hpp>
+#include <coap/coap_server.hpp>
+#include <common/timer.hpp>
+#include <meshcop/tlvs.hpp>
+#include <net/udp6.hpp>
+#include <thread/mle.hpp>
+
+namespace Thread {
+namespace MeshCoP {
+
+OT_TOOL_PACKED_BEGIN
+class CommissioningData
+{
+public:
+    uint8_t GetLength(void) const {
```

Comment:
This method is used only once: [here](https://github.com/openthread/openthread/blob/master/src/core/meshcop/leader.cpp#L109), FTD build size will **increase** 32 bytes after moving it from head file to CPP file:

- keep it in head file as is:
```
$ arm-none-eabi-size output/cc2538/bin/ot-cli-ftd 
   text	   data	    bss	    dec	    hex	filename
 143668	   4208	  17404	 165280	  285a0	output/cc2538/bin/ot-cli-ftd
```
- move it to CPP file: 
```
$ arm-none-eabi-size output/cc2538/bin/ot-cli-ftd 
   text	   data	    bss	    dec	    hex	filename
 143700	   4208	  17404	 165312	  285c0	output/cc2538/bin/ot-cli-ftd
```
So I think we should keep it as is, please let me know if you have any other concerns.